CHANGES/597.feature

```diff
@@ -0,0 +1 @@
+Made the context layer independent of click to allow it being used like a library.
```

Comment:
What's the next step in the plan? Move into separate repository or have other projects inherit from the CLI?

Reply:
That **is** the right question to ask. As for this change alone, it is a step towards better architectural separation of concerns. The next step i plan is to see if i can remove `echo` and `prompt` from `PulpContext`.
After that, it is either of:
1. Keep everthing as is, and install e.g. click when all you want is the library part. (unnecessary potential for version conflicts)
2. Make click and others an optional dependency of the cli. (kind of weird to install `pulp-cli[cli]`...)
3. Merge the cli-plugins and split into two repos/pypi packages.
4. Split the cli into 6+ repos/packages.

I'm really not ready to decide.